pyuvdata/uvdata/mir_parser.py

```diff
@@ -225,696 +42,1062 @@ class MirParser(object):
+                        continue
+                    int_size = rec_dict["record_size"]
+                    int_start = rec_dict["record_start"]
+                if (int_size != last_size) or (
+                    last_offset + (8 + last_size) * num_vals != int_start
                 ):
                     # Numpy's fromfile works fastest when reading multiple instances
                     # of the same dtype. As long as the record sizes are the same, we
                     # can tie mutiple file reads together into one call. The dtype
                     # depends on the record size, which is why we have to dump the
                     # data when last_size changes.
                     if num_vals != 0 and last_size != 0:
-                        in_data_dict.update(
-                            zip(
-                                inhid_list,
-                                np.fromfile(
-                                    visibilities_file,
-                                    dtype=in_dtype_dict[last_size],
-                                    count=num_vals,
-                                    offset=del_offset,
-                                ).copy(),
-                            )
+                        read_list.append(
+                            {
+                                "inhid_list": inhid_list,
+                                "int_dtype_dict": int_dtype_dict[last_size],
+                                "num_vals": num_vals,
+                                "del_offset": del_offset,
+                                "start_offset": last_offset,
+                            }
                         )
-                    del_offset = in_start - (last_offset + (num_vals * last_size))
-                    last_offset = in_start
-                    last_size = in_size
+                    # Capture the difference between the last integration and this
+                    # integration that we're going to drop into the next read.
+                    del_offset = int_start - (last_offset + (num_vals * last_size))
+                    # Starting positino for a sequence of integrations
+                    last_offset = int_start
+                    # Size of record (make sure all records are the same size in 1 read)
+                    last_size = int_size
+                    # Number of integraions in the read
                     num_vals = 0
+                    # Tally all the inhids in the read
                     inhid_list = []
                 num_vals += 1
                 inhid_list.append(ind_key)
-        return in_data_dict
+            filename = os.path.join(filepath, indv_file_dict[data_type]["filetype"])
+            # Time to actually read in the data
+            if use_mmap:
+                # kwargs = {
+                #     "mode": "r",
+                #     "offset": read_dict["start_offset"],
+                #     "shape": (read_dict["num_vals"],),
+                # }
```

Comment:
Is this still needed? If not I'd remove it.

Reply:
Removed in a2f46b4b.